ipn/ipnlocal/local.go

```diff
@@ -4729,7 +4730,34 @@ type tailFSTransport struct {
 	b *LocalBackend
 }
 
-func (t *tailFSTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+func (t *tailFSTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	bw := &requestBodyWrapper{}
+	if req.Body != nil {
+		bw.ReadCloser = req.Body
+		req.Body = bw
+	}
+
+	defer func() {
+		if req.Method != httpm.PUT && req.Method != httpm.GET {
+			return
+		}
+
+		i := strings.LastIndex(req.URL.String(), ".")
```

Comment:
Instead of using the extension, you can use the `Content-Type` header, which will be both more descriptive and more accurate.. On `GET`, you'll want to use the Content-Type from the `Response`, on PUT from the `Request`. On `PUT`, it'll won't always be available and you may need to fall back to the extension. Maybe we log both the content-type and file extension?